pkg/usage/versionset.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2020 The OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package usage
+
+import (
+	k8sapi "github.com/openebs/zfs-localpv/pkg/client/k8s/v1alpha1"
+	env "github.com/openebs/zfs-localpv/pkg/common/env"
+	openebsversion "github.com/openebs/zfs-localpv/pkg/version"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog"
+)
+
+var (
+	clusterUUID    = "OPENEBS_IO_USAGE_UUID"
+	clusterVersion = "OPENEBS_IO_K8S_VERSION"
+	clusterArch    = "OPENEBS_IO_K8S_ARCH"
+	openEBSversion = "OPENEBS_IO_VERSION_TAG"
+	nodeType       = "OPENEBS_IO_NODE_TYPE"
+	installerType  = "OPENEBS_IO_INSTALLER_TYPE"
+)
+
+// versionSet is a struct which stores (sort of) fixed information about a
+// k8s environment
+type versionSet struct {
+	id             string // OPENEBS_IO_USAGE_UUID
+	k8sVersion     string // OPENEBS_IO_K8S_VERSION
+	k8sArch        string // OPENEBS_IO_K8S_ARCH
+	openebsVersion string // OPENEBS_IO_VERSION_TAG
+	nodeType       string // OPENEBS_IO_NODE_TYPE
+	installerType  string // OPENEBS_IO_INSTALLER_TYPE
+}
+
+// NewVersion returns a new versionSet struct
+func NewVersion() *versionSet {
+	return &versionSet{}
+}
+
+// fetchAndSetVersion consumes the Kubernetes API to get environment constants
+// and returns a versionSet struct
+func (v *versionSet) fetchAndSetVersion() error {
+	var err error
+	v.id, err = getUUIDbyNS("default")
+	if err != nil {
+		return err
+	}
+	env.Set(clusterUUID, v.id)
+
+	k8s, err := k8sapi.GetServerVersion()
+	if err != nil {
+		return err
+	}
+	// eg. linux/amd64
+	v.k8sArch = k8s.Platform
+	v.k8sVersion = k8s.GitVersion
+	env.Set(clusterArch, v.k8sArch)
+	env.Set(clusterVersion, v.k8sVersion)
+	v.nodeType, err = k8sapi.GetOSAndKernelVersion()
+	env.Set(nodeType, v.nodeType)
+	if err != nil {
+		return err
+	}
+	v.openebsVersion = openebsversion.GetVersionDetails()
```

Comment:
(optional) `zfsLocalpvVersion`

Aside, if the app-versions are made to match the OpenEBS releases, we are good, if they are not we might want to add a prefix such as `zfs_lpv`, so that the version `zfs_lpv-0.9.1` and OpenEBS `0.9.1` aren't mixed up together esp if the AppNames are kept same as `OpenEBS`.

cc @kmova opinions?

Reply:
added "zfs-" as prefix to the version tag.